src/postgres/commands/PostgresAccountWizard/PostgresServerNameStep.ts

```diff
@@ -46,8 +46,12 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
     const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
 
     if (!availability.nameAvailable) {
-
-        return availability.message;
+        if (availability.reason === 'AlreadyExists') {
+            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
+        } else if (!name.match(/^(?![-])[a-zA-Z0-9-]*(?<![-])$/)) {
```

Comment:
Put this check for special characters before you call `await client.checkNameAvailability.execute`. Remember these validate methods happen every keystroke, so we want to backload azure calls (which may take a while) as much as possible